Analysis/Tasks/PWGCF/GenericFramework.cxx

```diff
@@ -0,0 +1,195 @@
+    oba->Add(new TNamed("ChFull34","ChFull34")); //no-gap case
+    oba->Add(new TNamed("ChGap42","ChGap42")); //gap case
+    oba->Add(new TNamed("LM22","LM22")); //for gap (|eta|>0.4) case
+    oba->Add(new TNamed("MR22","MR22")); //for gap (|eta|>0.4) case
+    oba->Add(new TNamed("LR22","LR22")); //for gap (|eta|>0.4) case
+    oba->Add(new TNamed("LLMR24","LLMR24")); //for gap (|eta|>0.4) case
+    oba->Add(new TNamed("LMMR24","LMMR24")); //for gap (|eta|>0.4) case
+    oba->Add(new TNamed("LMRR24","LMRR24")); //for gap (|eta|>0.4) case
+    oba->Add(new TNamed("ChSC234","ChSC234")); //for SC{2,3}
+    oba->Add(new TNamed("ChSC244","ChSC244")); //for SC{2,4}
+    TString fcname("FlowContainer");
+    fFC->SetName(fcname.Data());
+    int fNMultiBins = 9;
+    double fMultiBins[] = {0.0,5.0,10.0,20.0,30.0,40.0,50.0,60.0,70.0,80.0};
+    fFC->Initialize(oba,fNMultiBins,fMultiBins,1);
+    delete oba;
+
+  }
+
+  bool FillFC(const GFW::CorrConfig &corconf, const double &cent, const double &rndm)
+  {
+    double dnx,val;
+    dnx = fGFW->Calculate(corrconfigs.at(0),0,kTRUE).Re();
+    if(dnx==0) return false;
+    if(!corconf.pTDif) {
+      val = fGFW->Calculate(corrconfigs.at(0),0,kFALSE).Re()/dnx;
+      if(TMath::Abs(val)<1) fFC->FillProfile(corrconfigs.at(0).Head.Data(),cent,val,1,rndm);
+      return true;
+    }
+    return true;
+  }
+
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::Cents>>::iterator const& collision, aod::BCsWithTimestamps const&, myTracks const& tracks)
+  {
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+
+    if (cfgAcceptance.value.empty() == false) {
+      cfg.mAcceptance = ccdb->getForTimeStamp<GFWWeights>(cfgAcceptance.value, bc.timestamp());
+      if(cfg.mAcceptance) LOGF(info, "Loaded acceptance histogram from %s (%p)", cfgAcceptance.value.c_str(), (void*)cfg.mAcceptance);
+      else LOGF(warning,"Could not load acceptance histogram from %s (%p)", cfgAcceptance.value.c_str(), (void*)cfg.mAcceptance);
+    }
+
+    if(tracks.size()<1) return;
+    double vtxz = collision.posZ();
+    registry.fill(HIST("hVtxZ"),vtxz);
+    fGFW->Clear();
+    const auto centrality = collision.centV0M();
+    double dnx, val;
+    double l_Random = fRndm->Rndm();
+    double weff, wacc;
+    for (auto track = tracks.begin(); track != tracks.end(); ++track) {
+      registry.fill(HIST("hPhi"),track.phi());
+      registry.fill(HIST("hEta"),track.eta());
+      if(cfg.mEfficiency) weff = cfg.mEfficiency->GetBinContent(cfg.mEfficiency->FindBin(track.pt()));
+      else weff = 1.0;
+      if(weff==0) continue;
+      weff = 1./weff;
+      if(cfg.mAcceptance) wacc = cfg.mAcceptance->GetNUA(track.phi(),track.eta(),vtxz);
+      else wacc = 1;
+
+      fGFW->Fill(track.eta(),1,track.phi(),wacc*weff,3);
+    } 
+    for(int l_ind=0; l_ind<corrconfigs.size(); l_ind++) {
+      bool filled = FillFC(corrconfigs.at(l_ind),centrality,l_Random);
```

Comment:
bool is not needed

Reply:
Thanks a lot for the comments! I have fixed the mistakes and created a dedicated folder in PWGCF for the project now